travis-build.bash

```diff
@@ -34,28 +34,120 @@ function main () {
+      if [[ $NPM_TOKEN ]] ; then
+          echo "I have an NPM token! It could happen!"
+          local current_module_name
+          current_module_name=$(jq --raw-output .name package.json)
+          if [[ $? -ne 0 || ! -e package.json ]]; then
+            err "failed to parse name in package.json"
+            return 1
+          fi
+          local branch_module_name
+          branch_module_name="${current_module_name}_$TRAVIS_PULL_REQUEST_BRANCH"
+
+          # update the package.json
+          local temp_package_json
+          temp_package_json=$(mktemp)
+          if [[ $? -ne 0 ]]; then
+            err "failed to get a temp file"
+            return 1
+          fi
+
+          mv package.json ${temp_package_json}
+          jq ".name=\"${branch_module_name}\"" ${temp_package_json} > package.json
+          if [[ $? -ne 0 || ! -e package.json ]]; then
+            err "failed to update name in package.json"
+            return 1
+          fi
+          rm ${temp_package_json}
+
+          # create this file so we can use npm show on private modules
+           msg "creating local .npmrc using NPM token from environment"
+            if ! ( umask 077 && echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" > "$HOME/.npmrc" ); then
+                err "failed to create $HOME/.npmrc"
                 return 1
             fi
-            local timestamp
-            timestamp=$(date -u +%Y%m%d%H%M%S)
-            if [[ $? -ne 0 || ! $timestamp ]]; then
-                err "failed to generate timestamp"
+            # tell the npm-publish script that we've done this already
+            unset NPM_TOKEN
+
+          # is there already one of these published ?
+          local last_existing_version
+          last_existing_version=$(npm show ${branch_module_name} version)
+          if [[ $? -ne 0 ]] ; then
+             # probably not
+             echo "Looks like this is the first time we've published this branch, cool"
+          else
+             # increment the version. First set us to the current, then bump.
+             mv package.json ${temp_package_json}
+             jq ".version=\"${last_existing_version}\"" ${temp_package_json} > package.json
+             if [[ $? -ne 0 || ! -e package.json ]]; then
+                err "failed to update version in package.json"
+                return 1
+             fi
+             rm ${temp_package_json}
+
+             if ! npm version -f patch ; then
+                err "failed to increment version in package.json"
                 return 1
+             fi
+          fi
+
+# what version did we pick?
+          local pkg_version
+          pkg_version=$(jq --raw-output .version package.json)
+          if [[ $? -ne 0 || ! $pkg_version ]]; then
+            err "failed to parse version from package.json"
+            return 1
+          fi
+
+          if ! bash npm-publish.bash ${pkg_version} restricted ; then
```

Comment:
`"$pkg_version"`

With the new `npm-publish.bash` script, this is no longer necessary.